src/lib/storage/table.cpp

```diff
@@ -1,89 +1,111 @@
 #include "table.hpp"
 
+#include "resolve_type.hpp"
 #include "utils/assert.hpp"
+#include "value_segment.hpp"
 
 namespace opossum {
 
-Table::Table(const ChunkOffset target_chunk_size) {
-  // Implementation goes here
+Table::Table(const ChunkOffset target_chunk_size) : _chunk_size{target_chunk_size} {
+  create_new_chunk();
 }
 
 void Table::add_column_definition(const std::string& name, const std::string& type, const bool nullable) {
-  // Implementation goes here
+  _column_names.emplace_back(name);
+  _column_types.emplace_back(type);
+  _is_column_nullable.emplace_back(nullable);
+}
+
+static void add_value_segment(Chunk& chunk, const std::string& type, const bool nullable) {
+  resolve_data_type(type, [&](const auto data_type_t) {
+    using ColumnDataType = typename decltype(data_type_t)::type;
+    const auto value_segment = std::make_shared<ValueSegment<ColumnDataType>>(nullable);
+    chunk.add_segment(std::move(value_segment));
+  });
 }
 
 void Table::add_column(const std::string& name, const std::string& type, const bool nullable) {
-  // Implementation goes here
-  Fail("Implementation is missing.");
+  Assert(row_count() == 0, "Tried to create column on non-empty table.");
+  add_column_definition(name, type, nullable);
+  add_value_segment(*_chunks.back(), type, nullable);
 }
 
 void Table::create_new_chunk() {
-  // Implementation goes here
+  _chunks.emplace_back(std::make_shared<Chunk>());
+  for (ColumnID id{0}; const auto& type : _column_types) {
```

Comment:
According to the styleguide, `auto to stick` and variable naming want to be honoured here :)